src/ceph-volume/ceph_volume/devices/lvm/zap.py

```diff
@@ -287,10 +297,17 @@ def zap_raw_device(self, device):
             self.zap_partition(Device('/dev/%s' % part_name))
 
         zap_device(device.path)
+        if self.args.replace_block:
+            disk._dd_write(device.path, 'CEPH_DEVICE_BEING_REPLACED')
```

Comment:
just curious why this case is needed and the other part of the code that calls the `write_replacement_header` function 
```
                replacement_args: Dict[str, bool] = {
                    'block': self.args.replace_block,
                    'db': self.args.replace_db,
                    'wal': self.args.replace_wal
                }
                if replacement_args.get(lv.tags.get('ceph.type'), False):
                    mlogger.info(f'Marking {self.parent_device} as being replaced')
                    self.write_replacement_header(self.parent_device)
```
doesn't cover it

Reply:
`zap_raw_device()` is mostly for raw-based OSDs, my first intent with this PR was to make it support both raw and lvm based osds but that would require a bunch of changes and refactor that I would prefer to address in a second step. I should probably remove this or add a comment inline.